### ISSUE TITLE
On wake from sleep, devices seen in GUI but not controllable

### ISSUE THREAD

**Issue by PiotrKrzyzek** [ckb-next/ckb-next#148]
# Devices information

Currently running Mac OS High Sierra on a old 2014-ish iMac.
K95 Plat connected directly to computer.
MM800 pad connected through the keyboard.
[ckb-next-dev-detect-report.gz](https://github.com/ckb-next/ckb-next/files/1905563/ckb-next-dev-detect-report.gz)

## Logs & crash reports

Oddly, none. The daemon didn't report any activity at all upon wake.

## What is the program's current behavior?

When the computer goes to sleep, the devices will sleep as well. The lights will go off and all that. Upon wake the devices go into the default 'party demo' mode. The devices are seen in the GUI but are not controllable.

To restore functionality, one of the following things can be done:
- Unplug & replug in the devices
- Unload daemon via launchd, load daeom via launchd (essentially reinitialize daemon)
- Poke daemon to wake it up
- (maybe) Quit ckb-next gui (not just close it, but actually quit), re-open gui

## What is the expected behavior of the program?

When the computer goes to sleep, the devices will sleep as well. The lights will go off and all that. Upon wake the devices will come back on with the previous (pre-sleep) configurations and modes. The devices should be configurable.

## Any other notes & comments?

What is the "pipe command" referenced above? In the IRC chat we were chatting about allowing for command line configuration of colors to allow for external app activation notifications (such as, if new email, blink something). So, my theory is all that needs to happen is you somehow just need to 'ping' the ckb-next daemon to tell it "hey, wake up and pay attention" and in my case this new pipe command does the trick well. I just tell it to set to spacebar to nothing. Currently trying to figure out how to run a command on wake though. That's a weird one on Macs.

**Comment by PiotrKrzyzek** [comment]
This is not a solution, but a workaround that works:

Install the small Mac app called "sleepwatcher". It's available on HomeBrew if you use that.

1. Install and setup sleepwatcher (if installed via brew, remember to do `brew services start sleepwatcher`)
2. Using the new 'pipe' animation, create one for whatever key. Doesn't matter. I set it to the spacebar. 
3. Create a ".wakeup" file in your home folder with the following code:
`#!/bin/bash
echo ff00ff00 > /tmp/ckbpipe001`

Since the pipe command accepts RGBA (a being opacity), we simply set it to whatever color (in this case a purple) but the '00' at the end will essentially mean don't show the color we just set. Thus, it acts as a small 'ping' command to the ckb-next daemon to say "hey, wake up and pay attention".

Tested multiple times so far, and my keyboard and pad 'wakeup' after my computer does.

**Comment by tatokis** [comment]
This was fixed some time ago

**Comment by RobRoyce** [comment]
Forgive me if this question is misplaced, but I figured this was close enough that I shouldn't open a new issue.

My ST100 is connected to a USB hub -> monitor -> MBP. I ensured that my monitor does _not_ supply power during sleep mode. For some reason, however, the ST100 lights stay on even during sleep. They _occasionally_ turn off for some unspecified amount of time, but they typically stay on. It also fluctuates from the manufacture default "rainbow" mode to my CKB presets. 

It seems that OP has a Mac and his device is sleeping when the Mac sleeps, as should be the case. Is there anything I can try to get this thing to sleep when my MBP is asleep?
